html/pfappserver/lib/pfappserver/Form/Config/Source/Email.pm

```diff
@@ -51,6 +51,26 @@ has_field 'activation_domain' =>
     },
   );
 
+has_field 'allowed_domains' =>
+  (
+    type  => 'TextArea',
+    label => 'Comma-separated list of Allowed Domains',
+    tags => {
+        after_element => \&help,
+        help => 'A comma-separated list of domains that are allowed for email registration. Allowed domains are checked after banned domains.',
+    },
+  );
+
+has_field 'banned_domains' =>
+  (
+    type  => 'TextArea',
+    label => 'Comma-separated list of Banned Domains',
+    tags => {
+        after_element => \&help,
+        help => 'A comma-separated list of domains that are banned for email registration. Banned domains are checked before allowed domains.'
+    },
```

Comment:
if banned domains are checked before allowed domains, I propose to reverse order of fields in the form.

Reply:
If we support regex in domains fields, we should update "help" statements.